idb/postgres/postgres.go

```diff
@@ -1428,7 +1428,10 @@ func (db *IndexerDb) GetBlock(ctx context.Context, round uint64, options idb.Get
 			return types.Block{}, nil, err
 		}
 
-		go db.yieldTxnsThreadSimple(ctx, rows, out, true, nil, nil)
+		go func() {
+			db.yieldTxnsThreadSimple(ctx, rows, out, nil, nil)
+			close(out)
```

Comment:
I'm not sure about this pattern, seems like an accident waiting to happen when someone calls the function but forgets to call close afterwards.

Reply:
I think the worst thing that can happen is, you will get stuck. Tests would catch it.